src/xmipp/libraries/data/pdb.cpp

```diff
@@ -501,52 +515,79 @@ void PDBRichPhantom::read(const FileName &fnPDB, double pseudoatoms, double thre
 			// Extract atom type and position
 			// Typical line:
 			// ATOM    909  CA  ALA A 161      58.775  31.984 111.803  1.00 34.78
-			atom.name=line.substr(12,4);
-			atom.atomType = line[13];
-			atom.altloc=line[16];
-			atom.resname=line.substr(17,3);
-			atom.chainid=line[21];
-			atom.resseq = textToInteger(line.substr(22,4));
+			// ATOM      2  CA AALA A   1      73.796  56.531  56.644  0.50 84.78           C
+			atom.record = line.substr(0,6);
+			hy36decodeSafe(5, line.substr(6,5).c_str(), 5, &atom.serial);
+			atom.name = line.substr(12,4);
+			atom.altloc = line[16];
+			atom.resname = line.substr(17,3);
+			atom.chainid = line[21];
+			hy36decodeSafe(4, line.substr(22,4).c_str(), 4, &atom.resseq);
 			atom.icode = line[26];
 			atom.x = textToFloat(line.substr(30,8));
 			atom.y = textToFloat(line.substr(38,8));
 			atom.z = textToFloat(line.substr(46,8));
 			atom.occupancy = textToFloat(line.substr(54,6));
 			atom.bfactor = textToFloat(line.substr(60,6));
-			if(pseudoatoms != -1)
-				intensities.push_back(atom.bfactor);
+			atom.segment = line.substr(72,4);
+			atom.atomType = line.substr(76,2);
+			atom.charge = line.substr(78,2);
 
-			if(atom.bfactor >= threshold && pseudoatoms == -1)
+			if(pseudoatoms == true)
+				intensities.push_back(atom.bfactor);
+            
+			if(atom.bfactor >= threshold && pseudoatoms == false)
```

Comment:
`if(!pseudoatoms && atom.bfactor >= threshold)`
C++ has lazy condition evaluation, so you want the easy stuff first